tox.ini

```diff
@@ -84,27 +78,40 @@ setenv =
     {[base]setenv}
 changedir = {[base]changedir}
 commands =
-    pytest \
-        --durations=5 \
-        --cov={[base]covtarget} \
-        --cov-report=html:htmlcov-py37 --cov-report=term \
-        {posargs} \
-        {[base]pytesttarget}
+    pytest --durations=5 \
+           --cov={[base]covtarget} \
+           --cov-report=html:htmlcov-py37 \
+           --cov-report=term \
+           {posargs} \
+           {[base]pytesttarget}
 
 [testenv:py38]
+basepython = python3.8
 passenv = {[base]passenv}
 setenv =
     {[base]setenv}
 changedir = {[base]changedir}
-basepython = python3.8
 commands =
-    pytest \
-        --durations=5 \
-        --cov={[base]covtarget} \
-        --cov-report=html:htmlcov-py38 --cov-report=term \
-        {posargs} \
-        {[base]pytesttarget}
+    pytest --durations=5 \
+           --cov={[base]covtarget} \
+           --cov-report=html:htmlcov-py38 \
+           --cov-report=term \
+           {posargs} \
+           {[base]pytesttarget}
 
+[testenv:black]
+basepython = python3.6
+deps =
+    black
+commands =
+    black --check --diff --include "^[^.].*\.py$" --exclude "/(\.[^.].*|tests/roles)/" .
```

Comment:
can't we recurse in to subdirectories? this will avoid the nned of listing the subdirectories explicitly in `LSR_PYLINT_DIRS`

Reply:
In fact, we can write a custom `init-hook` in `pylintrc` which can do all the necessary work behind directory probing for us (including handling the special cases) and then abandon `LSR_PYLINT_DIRS` completely.

Reply:
sounds good

Reply:
I made a wrapper around `pylint`. Details in commit f21d01a.